Bug in validating and defaulting cluster endpoint cluster in `create cluster -f`

**What happened?**

config file:
```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: test-endpoint
  region: ap-southeast-2
iam:
  serviceRolePermissionsBoundary: "arn:aws:iam::XXX:policy/cfn-deploy-boundary-policy"
vpc:
  subnets:
    private:
      ap-southeast-2a:
        id: "subnet-XXX"
      ap-southeast-2b:
        id: "subnet-XXX"
      ap-southeast-2c:
        id: "subnet-XXX"
    public:
      ap-southeast-2a:
        id: "subnet-XXX"
      ap-southeast-2b:
        id: "subnet-XXX"
      ap-southeast-2c:
        id: "subnet-XXX"
managedNodeGroups:
  - name: standard-workers
    instanceType: t3.medium
#    privateNetworking: true
    desiredCapacity: 2
    minSize: 2
    maxSize: 10
```

```
./eksctl create cluster --cfn-role-arn arn:aws:iam::123:role/asd -f test.yaml
[ℹ]  eksctl version 0.14.0-rc.0
[ℹ]  using region ap-southeast-2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1487ba0]
goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5.NoAccess(...)
        github.com/weaveworks/eksctl@/pkg/apis/eksctl.io/v1alpha5/validation.go:125
github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5.(*ClusterConfig).ValidateClusterEndpointConfig(0xc00067e210, 0xc0004ba360, 0x0)
        github.com/weaveworks/eksctl@/pkg/apis/eksctl.io/v1alpha5/validation.go:114 +0x40
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc0002dc080, 0xc0005943c0, 0xc0002dc100, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:112 +0x168
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc0002c2500, 0xc00066a300, 0x0, 0x4, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:36 +0x9b
github.com/spf13/cobra.(*Command).execute(0xc0002c2500, 0xc00066a2c0, 0x4, 0x4, 0xc0002c2500, 0xc00066a2c0)
        github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002c2000, 0xc0002ed9f0, 0x1, 0x1)
        github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        github.com/weaveworks/eksctl@/cmd/eksctl/main.go:79 +0x407
```


## eksctl version 
`0.14.0-rc.0`

(Submitted through slack by @sayboras )

--- a/pkg/apis/eksctl.io/v1alpha5/defaults.go
+++ b/pkg/apis/eksctl.io/v1alpha5/defaults.go
@@ -178,17 +178,17 @@ func DefaultClusterNAT() *ClusterNAT {
 }
 
 // SetClusterEndpointAccessDefaults sets the default values for cluster endpoint access
-func SetClusterEndpointAccessDefaults(endpoints *ClusterEndpoints) {
-	if endpoints == nil {
-		endpoints = ClusterEndpointAccessDefaults()
+func SetClusterEndpointAccessDefaults(vpc *ClusterVPC) {
+	if vpc.ClusterEndpoints == nil {
+		vpc.ClusterEndpoints = ClusterEndpointAccessDefaults()
 	}
 
-	if endpoints.PublicAccess == nil {
-		endpoints.PublicAccess = Enabled()
+	if vpc.ClusterEndpoints.PublicAccess == nil {
+		vpc.ClusterEndpoints.PublicAccess = Enabled()
 	}
 
-	if endpoints.PrivateAccess == nil {
-		endpoints.PrivateAccess = Disabled()
+	if vpc.ClusterEndpoints.PrivateAccess == nil {
+		vpc.ClusterEndpoints.PrivateAccess = Disabled()
 	}
 }
 

--- a/pkg/ctl/cmdutils/configfile.go
+++ b/pkg/ctl/cmdutils/configfile.go
@@ -192,7 +192,7 @@ func NewCreateClusterLoader(cmd *Cmd, ngFilter *NodeGroupFilter, ng *api.NodeGro
 			*l.ClusterConfig.VPC.NAT.Gateway = api.ClusterSingleNAT
 		}
 
-		api.SetClusterEndpointAccessDefaults(l.ClusterConfig.VPC.ClusterEndpoints)
+		api.SetClusterEndpointAccessDefaults(l.ClusterConfig.VPC)
 
 		if !l.ClusterConfig.HasClusterEndpointAccess() {
 			return api.ErrClusterEndpointNoAccess

--- a/pkg/ctl/cmdutils/configfile_test.go
+++ b/pkg/ctl/cmdutils/configfile_test.go
@@ -260,5 +260,50 @@ var _ = Describe("cmdutils configfile", func() {
 			}
 
 		})
+
+		Describe("should set defaults for cluster endpoint access", func() {
+
+			testClusterEndpointAccessDefaults := func(configFilePath string, expectedPrivAccess, expectedPubAccess bool) {
+				cmd := &Cmd{
+					CobraCommand:      newCmd(),
+					ClusterConfigFile: configFilePath,
+					ClusterConfig:     api.NewClusterConfig(),
+					ProviderConfig:    &api.ProviderConfig{},
+				}
+
+				params := &CreateClusterCmdParams{
+					WithoutNodeGroup: true,
+					Managed:          false,
+				}
+
+				Expect(NewCreateClusterLoader(cmd, NewNodeGroupFilter(), nil, params).Load()).To(Succeed())
+				cfg := cmd.ClusterConfig
+				assertValidClusterEndpoint(cfg.VPC.ClusterEndpoints, expectedPrivAccess, expectedPubAccess)
+			}
+
+			It("when VPC is imported and no access is defined", func() {
+				testClusterEndpointAccessDefaults("test_data/cluster-with-vpc.yaml", false, true)
+			})
+
+			It("when VPC is created by eksctl and no access is defined", func() {
+				testClusterEndpointAccessDefaults("test_data/cluster-without-vpc.yaml", false, true)
+			})
+
+			It("when VPC is created by eksctl and private endpoint is enabled", func() {
+				testClusterEndpointAccessDefaults("test_data/cluster-without-vpc-private-access.yaml", true, true)
+			})
+
+			It("when VPC is imported and private endpoint is enabled", func() {
+				testClusterEndpointAccessDefaults("test_data/cluster-with-vpc-private-access.yaml", true, true)
+			})
+		})
 	})
 })
+
+func assertValidClusterEndpoint(endpoints *api.ClusterEndpoints, privateAccess, publicAccess bool) {
+	Expect(endpoints).To(Not(BeNil()))
+	Expect(endpoints.PrivateAccess).To(Not(BeNil()))
+	Expect(*endpoints.PrivateAccess).To(Equal(privateAccess))
+	Expect(endpoints.PublicAccess).To(Not(BeNil()))
+	Expect(*endpoints.PublicAccess).To(Equal(publicAccess))
+}

--- a/pkg/ctl/cmdutils/test_data/cluster-with-vpc-private-access.yaml
+++ b/pkg/ctl/cmdutils/test_data/cluster-with-vpc-private-access.yaml
@@ -1,0 +1,27 @@
+# A simple example of ClusterConfig object:
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: test-cluster-1
+  region: eu-north-1
+
+vpc:
+  id: "vpc-12345"
+  cidr: "10.0.0.0/16"
+  subnets:
+    private:
+      us-west-1a:
+        id: "subnet-12345"
+        cidr: "10.0.1.0/24"
+      us-west-2b:
+        id: "subnet-67890"
+        cidr: "10.0.2.0/24"
+  clusterEndpoints:
+    privateAccess: true
+
+nodeGroups:
+  - name: ng-1
+    instanceType: m5.large
+    desiredCapacity: 1

--- a/pkg/ctl/cmdutils/test_data/cluster-with-vpc.yaml
+++ b/pkg/ctl/cmdutils/test_data/cluster-with-vpc.yaml
@@ -1,0 +1,26 @@
+# A simple example of ClusterConfig object:
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: test-cluster-1
+  region: eu-north-1
+
+vpc:
+  id: "vpc-12345"
+  cidr: "10.0.0.0/16"
+  subnets:
+    private:
+      us-west-1a:
+        id: "subnet-12345"
+        cidr: "10.0.1.0/24"
+      us-west-2b:
+        id: "subnet-67890"
+        cidr: "10.0.2.0/24"
+
+
+nodeGroups:
+  - name: ng-1
+    instanceType: m5.large
+    desiredCapacity: 1

--- a/pkg/ctl/cmdutils/test_data/cluster-without-vpc-private-access.yaml
+++ b/pkg/ctl/cmdutils/test_data/cluster-without-vpc-private-access.yaml
@@ -1,0 +1,17 @@
+# A simple example of ClusterConfig object:
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: test-cluster-1
+  region: eu-north-1
+
+vpc:
+  clusterEndpoints:
+    privateAccess: true
+
+nodeGroups:
+  - name: ng-1
+    instanceType: m5.large
+    desiredCapacity: 1

--- a/pkg/ctl/cmdutils/test_data/cluster-without-vpc.yaml
+++ b/pkg/ctl/cmdutils/test_data/cluster-without-vpc.yaml
@@ -1,0 +1,13 @@
+# A simple example of ClusterConfig object:
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: test-cluster-1
+  region: eu-north-1
+
+nodeGroups:
+  - name: ng-1
+    instanceType: m5.large
+    desiredCapacity: 1

